### ISSUE TITLE
Query context / intention

### ISSUE THREAD

**Issue by brettasmi** [NCATSTranslator/ReasonerAPI#319]
It would be useful to be able to encode a user's "intention" or "context" in the query graph. An ARA might want to use this information to rank results, and a KP might be able to return results specific to a context.

How should this parameter be encoded in the query graph? A simple parameter, e.g. `context`, with an array of CURIEs could be appropriate. However, qualifiers could be used on an edge-by-edge basis to guide the KP behavior.

Also, there are some discussions ongoing in the data modeling group that cover "context," but @mbrush confirms that they are lower priority for the time being.

Because this issue touches multiple working groups, it has been [elevated to the architecture group](https://github.com/NCATSTranslator/TranslatorArchitecture/issues/69) as well. Any implementation here should take into account discussions in that group.

**Comment by edeutsch** [comment]
It seems that many discussions have led to the recommendation that ARAs could look for an honor additional property Query.query_context that can contain a CURIE (or a list of CURIEs?) that specify a context for evaluating a Query and/or ranking the final results.

**Comment by brettasmi** [comment]
Unfortunately, we don't have an example to discuss for the "query_context" today. While I did write an example, we had to cancel our internal imProving meeting during which we planned to discuss it, so we should probably delay that discussion if possible.

We were able to have a minor discussion asynchronously that perhaps the term "context" isn't the best choice here, as this is intended to be a concept with which to rank (at least that was my original intention). As such, something like "ranking_hint" or "ranking_prompt" may be better for clarity.

**Comment by ehinderer** [comment]
I agree that in theory the "level" or "degree" of knowledge about a statement can help to provide relevancy to query results. The discussion in #324 gives a good overview of the kinds contexts that a user might be interested in. For instance if a user doesn't want to see "no brainer" answers, they might filter out "known to treat" and may be more interested in "indirectly inferred to treat." 

Before I can comment on about how to structure the query, I need to understand how to answer those kinds of questions, conceptually. For instance. what is meant by "inferred"? Are we expecting KPs to literally have edges/statements that amount to the phrase "indirectly inferred to" or are we expecting the ARAs to reason across "known to" statements and make the "indirectly inferred to" statement at that level? If its the former, then the query would constrain on QG edge types, if its the latter, then the query would specify sophisticated ARA operations. Maybe it's a combination of the two?
